### ISSUE TITLE
Banning based on forwarded IP

### ISSUE THREAD

**Issue by stashymane** [tomMoulard/fail2ban#84]
As noted in [this discussion](https://github.com/tomMoulard/fail2ban/discussions/83), fail2ban currently only uses the request's IP for banning. This makes it impossible to properly ban IPs when behind a proxy, like Cloudflare.
A configuration option to pull the IP from a request header or other source would be ideal.

**Comment by fliot** [comment]
100% agree on this very important feature.

**Comment by zhangjunjie6b** [comment]
I have to turn off the proxy function of Cloudflare for the same issue

**Comment by johnbizokk** [comment]
Indeed! Without support for blocking the IP address from the x-forwarded-for header, the plugin is non-functional when the traefik is behind a proxy (such as Cloudflare).

**Comment by tomMoulard** [comment]
Indeed this could be a very important feature. It would be nice to have a configuration like the [source criterion](https://doc.traefik.io/traefik/middlewares/http/inflightreq/#sourcecriterion) of the InFlightReq middleware.
I'll be waiting for a PR until I'll do it myself in a while.
